### ISSUE TITLE
callback-return does not work with object methods

### ISSUE THREAD

**Issue by fakewaffle** [eslint/eslint#4711]
I've read the documentation for the [rule](http://eslint.org/docs/rules/callback-return), but did not see anything related to this; nor could I find an issue.

**.eslintrc**
```json
{
	"env" : {
		"node" : true
	},

	"rules" : {
		"callback-return" : [ 2, [ "callback", "cb", "done", "next", "send.error", "send.success", "send.fail", "reply" ] ]
	}
}
```

**callback-return.js**

```js
'use strict'

( function ( item, send, callback ) {
	if ( item ) {
		send.fail();
	}

	send.success();

	if ( item ) {
		callback( item );
	}

	callback();
} );
```

**console**

```
~/Source/temp
> eslint -c .eslintrc callback-return.js

~/Source/temp/callback-return.js
  11:3  error  Expected return with your callback function  callback-return

✖ 1 problem (1 error, 0 warnings)
```

Given the options I supplied to the rule, I would expect ESLint to give an error for `send.fail();`.

Thoughts?

**Comment by gyandeeps** [comment]
Based on the call it only works for `callee as Identifier` and not for `callee as MemberExpression`.

@eslint/eslint-team Thoughts?

**Comment by nzakas** [comment]
I'm open to this enhancement.

@fakewaffle do you want to submit a pull request?

**Comment by fakewaffle** [comment]
@gyandeeps `v1.10.3`.

@nzakas Yes!

**Comment by gyandeeps** [comment]
@fakewaffle I can see u have done implementation but you havent opened a pull request. I dont know if you forgot to that or is deliberate. Just a reminder. :smile:

**Comment by fakewaffle** [comment]
@gyandeeps #4716?

**Comment by josectello** [comment]
If no one is currently working on this, I'd love to give it a shot and pickup where @fakewaffle left off.

**Comment by ilyavolodin** [comment]
@josectello Go for it!
